src/perform_step/bdf_perform_step.jl

```diff
@@ -454,3 +454,129 @@ function perform_step!(integrator,cache::QNDF2Cache,repeat_step=false)
   cache.newton_iters = iter
   f(integrator.fsallast, u, p, t+dt)
 end
+
+function initialize!(integrator, cache::QNDFConstantCache)
+  integrator.kshortsize = 2
+  integrator.k = typeof(integrator.k)(integrator.kshortsize)
+  integrator.fsalfirst = integrator.f(integrator.uprev, integrator.p, integrator.t) # Pre-start fsal
+
+  # Avoid undefined entries if k is an array of arrays
+  integrator.fsallast = zero(integrator.fsalfirst)
+  integrator.k[1] = integrator.fsalfirst
+  integrator.k[2] = integrator.fsallast
+end
+
+function perform_step!(integrator,cache::QNDFConstantCache,repeat_step=false)
+  @unpack t,dt,uprev,u,f,p = integrator
+  @unpack udiff,dts,k,max_order,D,D2,R,U = cache
+  cnt = integrator.iter
+  κ = integrator.alg.kappa[k]
+  γ = inv((1-κ)*γₖ[k])
+  flag = true
+  for i in 2:k
+    if dts[i] != dts[1]
+      flag = false
+      break
+    end
+  end
+  if cnt > 2
+    if flag
+      ρ = dt/dts[1]
+      # backward diff
+      for i = 1:k
+        D2[1,i] = udiff[i]
+      end
+      backward_diff(D,D2,k)
+      if ρ != 1
+        U!(k,U)
+        R!(k,ρ,cache)
+        D .= D * (R * U)
+      end
+    else
+      for i = 1:k
+        D2[1,i] = udiff[i] * dt/dts[i]
+      end
+      backward_diff(D,D2,k)
+    end
+  else
+    γ = 1//1
+  end
+  # precalculations
+  u₀ = uprev + sum(D)
+  ϕ = zero(γ)
+  for i = 1:k
+    ϕ += γₖ[i]*D[i]
+  end
+  ϕ *= γ
+  tmp = u₀ - ϕ
+  γdt = γ*dt
+  W = calc_W!(integrator, cache, γdt, repeat_step)
+  # initial guess
+  z = dt*integrator.fsalfirst
```

Comment:
`u0` is the predictor point, so `dt*f(u0,p,t+dt)` should be the initial guess? @YingboMa can we get a double check here?

Reply:
it increases one more function evaluation but setting guess to `dt*f(u0,p,t+dt)` speed up convergence.

Reply:
Well it's a priori one more function evaluation if the Newton solve takes the same number of iterations. However, that is the Adams-Moulton estimate while `z = dt*integrator.fsalfirst` is just constant extrapolation (guessing that the derivative in the future is the derivative of the past) so this should overall reduce the number of function evaluations. Just need to make sure the formula is correct since our Newton setup isn't the "naive" version and instead we iterate `z = dt*f`, so the translation should be what I have above (and it would need one extra step to support mass matrices IIRC).